intellij/src/saros/intellij/eventhandler/filesystem/LocalFilesystemModificationHandler.java

```diff
@@ -577,10 +598,10 @@ private void generateFolderMoveActivity(
 
     Set<IProject> sharedReferencePoints = session.getProjects();
 
-    IFolder oldFolderWrapper =
-        (IFolder) VirtualFileConverter.convertToResource(sharedReferencePoints, oldFile);
-    IFolder newParentWrapper =
-        (IFolder) VirtualFileConverter.convertToResource(sharedReferencePoints, newParent);
+    IContainer oldFolderWrapper =
```

Comment:
What is your current idea concerning the interface relation of IContainer and IFolder?
Why using IContainer here instead of IFolder?

Reply:
> What is your current idea concerning the interface relation of IContainer and IFolder?

I am not sure. I thought about combining them after moving `create()` to `IResource()`, but it hasn't been a high priority.

> Why using IContainer here instead of IFolder?

This is due to the recent change to the patch set #997 disallowing resource instances having an empty path. As part of those changes, the logic of `VirtualFileConverter` was adjusted to return the reference point object if the method `covertToResource(...)` is called with a virtual file matching the one held in the given reference point. (This case was the reason for the change. Previously, this was handled by creating an `IFolder` with an empty path.)

As a result, the returned value can either be an `IFolder` or an `IReferencePoint` object. `IContainer` is the nearest shared parent interface. Furthermore, these variables are only used for the initial check whether the resources are shared, so it doesn't really matter. The actual `IFolder` instances that are used for the activity creation are located in the content iterator later in the method. (This works as the iterator includes the given base file.)